integration_test/README.md

```diff
@@ -142,7 +142,39 @@ Then, inside `applications/<application>/`:
 1.  (if necessary) `exercise`. This is only needed
     sometimes, e.g. to get the application to log to a particular file.
 1.  (if you want to test logging) `expected_logs.yaml`
-1.  (if you want to test metrics) `metric_name.txt`
+1.  (if you want to test metrics) `expected_metrics.yaml`
+
+### expected_metrics.yaml
+
+We use `expected_metrics.yaml` both as a test artifact and as a source for documentation. All metrics ingested from the integration should be documented here.
+
+A sample `expected_metrics.yaml` snippet looks like:
+
+```yaml
+- type: workload.googleapis.com/apache.current_connections
+  value_type: INT64
+  kind: GAUGE
+  monitored_resource: gce_instance
+  labels:
+    server_name: .*
+  is_representative: true
+  optional: true
+```
+
+`type`, `value_type` and `kind` come directly from the metric descriptor for that metric. `monitored_resource` should always be `gce_instance`.
+
+`labels` is an exhaustive list of labels associated with the metric. Each key in `labels` is the label name, and its value is a regular expression. During the test, each label returned by the time series for that metric is checked against `labels`: every label in the time series must be present in `labels`, and its value must match the regular expression.
+
+For example, if a metric defines a label `operation` whose values can only be `read` or `write`, then an appropriate `labels` map in `expected_metrics.yaml` would be as follows:
+
+```yaml
+  labels:
+    state: (read|write)
+```
+
+`is_representative` is optional. Exactly one metric from each integration's `expected_metrics.yaml` must have `is_representative: true`. This is used for documentation only, however the test will still validate that exactly one metric sets it to `true`.
+
+`optional` is optional. When set to `true`, the metric will be skipped during the test. This is useful for documenting metrics that we don't have tests for.
```

Comment:
> This is useful for documenting metrics that we don't have tests for.

Actually, this is useful for metrics that aren't always sent by the integration. Wasn't the plan to still verify this metric if it's present? The only difference here would be that, unlike for non-optional metrics, this metric's absence wouldn't fail the test.

Reply:
If that was the plan then I'm happy to change it. I did it this way to avoid extending the test runtime. Metrics seem to only be missing if the necessary steps to send them are unimplemented in the setup scripts, which means they'll be deterministically missing and the test will always time out. Not sure if "flaky metrics" are a thing though.

Reply:
They're not "flaky", as much as there may be some metrics that are present on some platforms but not others, and you don't want to make this file platform-specific…

Reply:
Updated behaviour & wording. The test will now check for `optional` metrics and pass if they are missing.